src/System.Linq/src/System/Linq/Partition.cs

```diff
@@ -180,7 +188,7 @@ public IPartition<TElement> Skip(int count)
         public IPartition<TElement> Take(int count)
         {
             int maxIndex = _minIndex + count - 1;
-            if (maxIndex >= _maxIndex) maxIndex = _maxIndex;
+            if ((uint)maxIndex >= (uint)_maxIndex) maxIndex = _maxIndex;
```

Comment:
If `_minIndex + count - 1` overflows then `maxIndex` will be negative, and should be replaced with `_maxIndex`.

In a way this is the opposite to the deliberate overflow of my preferred index check, in that we really do want to see if `maxIndex` was set to a value greater than `MaxValue`, and undo the overflow that would have caused when doing the check.